frontend/scripts/buildCache.ts

```diff
@@ -51,41 +55,104 @@ async function testAPIAccess() {
   }
 }
 
-function cacheRPCensus() {
-  return seq(...DATASETS.map(_ => async () => {
-    console.log('  Getting RPs and Census for ' + safeDatasetHint(_))
-    let representativePoints = await getRepresentativePoints({ service_area_ids: _.serviceAreaIds })
-    let census = await getCensusData({ service_area_ids: _.serviceAreaIds })
-    if (uploadToS3) {
-      let rpParams = { Bucket: s3Bucket, Key: getS3Key(getStaticRPUrl(_)), Body: JSON.stringify(representativePoints), ContentType: 'application/json', ACL: 'public-read' }
-      let censusParams = { Bucket: s3Bucket, Key: getS3Key(getStaticDemographicsUrl(_)), Body: JSON.stringify(census), ContentType: 'application/json', ACL: 'public-read' }
-      s3.putObject(rpParams, s3Callback)
-      s3.putObject(censusParams, s3Callback)
-    }
-  }))
-}
+function cacheData() {
+  return seq(...DATASETS.map(dataset => async () => {
+    let states: State[] = dataset.usaWide && cacheUSAWide ? STATES.map(_ => _.shortName as State) : [dataset.state]
+    console.log(`Dealing with ${safeDatasetHint(dataset)}...`)
+    return seq(...states.map(state => async () => {
+      dataset.state = state
+      dataset.serviceAreaIds = []
+      dataset = inferServiceAreaIds(dataset)
+
+      console.log(`  Getting Rps and Census for ${safeDatasetHint(dataset)}`)
+      let representativePoints = await getRepresentativePoints({ service_area_ids: dataset.serviceAreaIds })
+      let census = await getCensusData({ service_area_ids: dataset.serviceAreaIds })
+      if (!representativePoints) {
+        throw Error('Empty response for RPs.')
+      }
 
-function cacheAdequacies() {
-  return seq(...DATASETS.map(dataset => async () =>
-    seq(...METHODS.map(method => async () => {
-      console.log('  Getting Adequacy for ' + safeDatasetHint(dataset) + ' for ' + method)
-      let adequacies = await getAdequacies({
-        method,
-        providers: dataset.providers.map((_, n) => ({ latitude: _.lat, longitude: _.lng, id: n })),
-        service_area_ids: dataset.serviceAreaIds,
-        dataset_hint: safeDatasetHint(dataset)
-      })
-      if (uploadToS3 && adequacies) {
-        let adequacyParams = { Bucket: s3Bucket, Key: getS3Key(getStaticAdequacyUrl(dataset, method)), Body: JSON.stringify(adequacies), ContentType: 'application/json', ACL: 'public-read' }
-        s3.putObject(adequacyParams, s3Callback)
+      if (!census) {
+        throw Error('Empty response for census.')
       }
+
+      let rpParams = { Bucket: s3Bucket, Key: getS3Key(getStaticRPUrl(dataset)) }
+      let censusParams = { Bucket: s3Bucket, Key: getS3Key(getStaticDemographicsUrl(dataset)) }
+
+      await uploadIFFNotExists(rpParams, JSON.stringify(representativePoints), 'Rps ' + safeDatasetHint(dataset), forceS3Upload)
```

Comment:
is this IFF unless you pass a parameter that says JK just upload anyways? doesn't seem very IFFy to me